README.md

```diff
@@ -16,48 +16,62 @@ repository is compatible with the official Fast Downward Git repository.
-  CLEANED_MERCURIAL_REPOSITORY is a location where the intermediate cleaned up Mercurial 
-  repository will be written to:
-
-    ./run-cleanup.sh MERCURIAL_REPOSITORY CLEANED_MERCURIAL_REPOSITORY
+  ORDERED_REPOSITORY is an intermediate repository that ensures that the history
+  contains all commits in the same order as the Fast Downward master repository.
+  The CLEANED_MERCURIAL_REPOSITORY is a location where the intermediate cleaned
+  up Mercurial repository will be written to:
+  
+    ./run-cleanup.sh MERCURIAL_REPOSITORY ORDERED_REPOSITORY CLEANED_MERCURIAL_REPOSITORY
     ./run-conversion.sh CLEANED_MERCURIAL_REPOSITORY CONVERTED_GIT_REPOSITORY \
         [--redirect-fast-export-stderr FILE]
 
-The scripts will automatically set up the required tools (a virtual
-environment with compatible versions of Mercurial and the fast-export tool
-https://github.com/frej/fast-export.git).
+  The scripts will automatically set up the required tools (a virtual
+  environment with compatible versions of Mercurial and the fast-export tool
+  https://github.com/frej/fast-export.git).
 
 ## Limitations
-- Multiple Mercurial heads with the same branch name are not supported. If your
-  repository has those, you will see
-  `Error: repository has at least one unnamed head: hg rXXX`.
-- If you have closed and merged a branch "subfeature" into a branch "feature"
-  and "feature" is not yet merged into "main", you might want to delete "subfeature"
-  branch from the resulting Git repository by running `git branch -D subfeature`.
+  - Multiple Mercurial heads with the same branch name are not supported. If your
+    repository has those, you will see
+    `Error: repository has at least one unnamed head: hg rXXX`.
+  - If you have closed and merged a branch "subfeature" into a branch "feature"
+    and "feature" is not yet merged into "main", you will receive:
+    `error: The branch 'BRANCH' is not fully merged.`
+    Don't worry. You might want to delete "subfeature"
+    branch from the resulting Git repository by running `git branch -D subfeature`.
 
 ## Warnings
-- Both scripts generate a lot of output on stdout and stderr. If you want
-  to analyze it, better redirect it into files.
-- The cleanup script generates repeated warnings about missing or invalid tags.
-  These are caused by moved or broken tags and can be ignored.
+  - The `run-cleanup.sh` and `run-conversion.sh` scripts generate a lot of output 
+    on stdout and stderr. If you want to analyze it, better redirect it into files.
+  - The cleanup script generates repeated warnings about missing or invalid tags.
+    These are caused by moved or broken tags and can be ignored.
 
+## Troubleshooting
+  If you have problems with the `run-all-steps.sh` script, try to run the steps
+  individually and carefully inspect the output of each step. Depending on your 
+  problems it might help to first pull the changes from 
```

Comment:
I would see it now as, we force the user to pull the changes in first and our pulling is just ordering the commits. therefore, I think this can be removed, but we could state the the user has to pull in the latest changes manually.